docs/FEATURES.md

```diff
@@ -77,5 +79,58 @@ These headers include simple functions for joining a container into a single str
 ## Visibility definitions and macros {#visibility-definitions-and-macros}
 `rcpputils/visibility_control.hpp` provides macros and definitions for controlling the visibility of class members. The logic was borrowed and then namespaced from [https://gcc.gnu.org/wiki/Visibility](https://gcc.gnu.org/wiki/Visibility).
 
-## Shared Libraries
-`rcpputils/shared_library.hpp` provides dynamically loads, unloads and get symbols from shared libraries at run-time.
\ No newline at end of file
+## Shared Libraries {#shared-libraries}
+The `rcpputils/shared_library.hpp` header provides utilities to dynamically load, unload, and get symbols from shared libraries at run-time.
+
+Example usage:
+```c++
+// Obtain platform-specific library name from the provided base name.
+const std::string library_name = rcpputils::get_platform_library_name("foo_lib");
+
+// The shared library is loaded in the constructor.
+auto library = std::make_shared<rcpputils::SharedLibrary>(library_name);
+
+// Check if a symbol exists in the shared library.
+// If so, obtain a pointer to the symbol.
+if(library -> has_symbol("example_symbol")) {
+    std::shared_ptr<void> symbol(library -> get_symbol("example_symbol"));
+    // Use shared library symbol pointer.
+}
+```
+
+## Process helpers {#process-helpers}
+The `rcpputils/process.hpp` header contains process utilities.
+
+Namely, this header provides the `rcpputils::get_executable_name()` function, which retrieves and returns the current program name as a string. It is a *thread-safe* function.
```

Comment:
I'm not sure I would say it is "thread-safe" here.  While that is true, this is duplicating the API documentation.

Reply:
True. I wasn't sure if it was particularly important for users to know. Removed in a908398.